idrstream/idr_stream.py

```diff
@@ -310,6 +315,61 @@ def clear_batch(self):
 
         self.logger.info("Temporary batch files cleared")
 
+    def add_batch_object_outlines(
+        self, batch_single_cell_df: pd.DataFrame
+    ) -> pd.DataFrame:
+        """
+        add object outlines to single cell data for current batch
+        necessary because we compile original single cell df with pycytominer
+        but pycytominer cannot append additional metadata (like object outlines)
+
+        Parameters
+        ----------
+        batch_single_cell_df : pd.DataFrame
+            original batch single cell df
+
+        Returns
+        -------
+        pd.DataFrame
+            new batch single cell df with object outlines appended
+        """
+
+        locations_save_path = pathlib.Path(f"{self.DP_project_path}/inputs/locations/")
+        new_batch_single_cell_df = []
+
+        # iterate over location files in order of plate map data in batch_single_cell_df
+        for DNA_image_path in batch_single_cell_df["Metadata_DNA"].unique():
```

Comment:
will this always be "Metadata_DNA"? If not, then maybe make this a default function arg

Reply:
Done in https://github.com/WayScience/IDR_stream/pull/5/commits/ecd3a752f7bd55dce958113c02bcee06b2b0154e